src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -761,90 +761,100 @@ SegmentCleaner::gc_reclaim_space_ret SegmentCleaner::gc_reclaim_space()
     [this, segment_id, pavail_ratio, start, end_paddr](
       auto &reclaimed,
       auto &runs) {
-    return repeat_eagain(
-      [this, &reclaimed, segment_id, &runs, end_paddr]() mutable {
-      reclaimed = 0;
-      runs++;
+    return ecb->with_transaction_intr(
+      Transaction::src_t::READ,
+      "get_backref_mappings",
+      [this, end_paddr](auto &t) {
+      return backref_manager.get_mappings(
+	t, *next_reclaim_pos, end_paddr
+      );
```

Comment:
IIUC this is under the assumption that all cleaner transactions -- trim_backref, clearner_trim, clean_reclaim -- will not run in parallel, so they won't conflict with each other in the current code.

Reply:
yes